modules/plan/src/main/PlanApi.scala

```diff
@@ -368,103 +371,73 @@ final class PlanApi(
-    stripeClient.createCustomer(user, data, plan) flatMap { customer =>
-      saveStripePatron(user, customer.id, data.freq) >>
-        setDbUserPlan(user, lila.user.Plan.start) >>
-        notifier.onStart(user) >>-
-        logger.info(s"Create ${user.username} customer $customer") inject customer
-    }
-
   private def saveStripePatron(user: User, customerId: CustomerId, freq: Freq): Funit =
-    userPatron(user) flatMap {
-      case None =>
-        patronColl.insert.one(
-          Patron(
-            _id = Patron.UserId(user.id),
-            stripe = Patron.Stripe(customerId).some,
-            lastLevelUp = DateTime.now
-          ).expireInOneMonth(!freq.renew)
+    userPatron(user) flatMap { patronOpt =>
+      val patron = patronOpt
+        .getOrElse(Patron(_id = Patron.UserId(user.id)))
+        .copy(
+          stripe = Patron.Stripe(customerId).some,
+          lastLevelUp = Some(DateTime.now)
         )
-      case Some(patron) =>
-        patronColl.update.one(
-          $id(patron.id),
-          patron
-            .copy(
-              stripe = Patron.Stripe(customerId).some
-            )
-            .removePayPal
-            .expireInOneMonth(!freq.renew)
-        )
-    } void
-
-  private def setCustomerPlan(
-      customer: StripeCustomer,
-      plan: StripePlan,
-      source: Source
-  ): Fu[StripeSubscription] =
-    customer.subscriptions.data.find(_.plan == plan) match {
-      case Some(sub) => fuccess(sub)
-      case None =>
-        customer.firstSubscription match {
-          case None      => stripeClient.createSubscription(customer, plan, source)
-          case Some(sub) => stripeClient.updateSubscription(sub, plan, source.some)
-        }
+        .removePayPal
+        .expireInOneMonth(!freq.renew)
+      patronColl.update.one($id(user.id), patron, upsert = true).void
+    }
+
+  private def saveStripeCustomer(user: User, customerId: CustomerId): Funit =
+    userPatron(user) flatMap { patronOpt =>
+      val patron = patronOpt
+        .getOrElse(Patron(_id = Patron.UserId(user.id)))
+        .copy(stripe = Patron.Stripe(customerId).some)
+      patronColl.update.one($id(user.id), patron, upsert = true).void
     }
 
-  private def userCustomerId(user: User): Fu[Option[CustomerId]] =
+  def userCustomerId(user: User): Fu[Option[CustomerId]] =
     userPatron(user) map {
       _.flatMap { _.stripe.map(_.customerId) }
     }
 
-  private def userCustomer(user: User): Fu[Option[StripeCustomer]] =
+  def userCustomer(user: User): Fu[Option[StripeCustomer]] =
     userCustomerId(user) flatMap {
       _ ?? stripeClient.getCustomer
     }
 
+  def getOrMakeCustomer(user: User, data: Checkout): Fu[StripeCustomer] =
+    userCustomer(user) getOrElse makeCustomer(user, data)
+
+  def makeCustomer(user: User, data: Checkout): Fu[StripeCustomer] =
+    stripeClient.createCustomer(user, data) map { customer =>
+      {
+        saveStripeCustomer(user, customer.id);
+        customer
+      }
```

Comment:
Here, `saveStripeCustomer`'s future isn't chained, and we might miss a failure. Consider:

```
    stripeClient.createCustomer(user, data) flatMap { customer =>
        saveStripeCustomer(user, customer.id) inject customer
    }
```

Reply:
Whoops. Good point. I'll keep that in mind going forward.